From 83b7f6457e5dca3ee2145e35f361f1520d24be42 Mon Sep 17 00:00:00 2001
From: Ashok Sidipotu <ashok.sidipotu@collabora.com>
Date: Fri, 14 Jan 2022 18:30:44 +0530
Subject: [PATCH 3/4] Revert "meson: declare spa_dep and override_dependency()
 for spa and pipewire"

This reverts commit 7ab5c35cadc075bcdf44031f1f7d95352cf9a13b.

Upstream-Status: Inappropriate[meson version dependent]
---
 meson.build              |  2 +-
 spa/meson.build          | 13 +------------
 src/pipewire/meson.build | 10 ++--------
 3 files changed, 4 insertions(+), 21 deletions(-)

diff --git a/meson.build b/meson.build
index 67e697c15..476f722b7 100644
--- a/meson.build
+++ b/meson.build
@@ -1,7 +1,7 @@
 project('pipewire', ['c' ],
   version : '0.3.43',
   license : [ 'MIT', 'LGPL-2.1-or-later', 'GPL-2.0-only' ],
-  meson_version : '>= 0.56.0',
+  meson_version : '>= 0.49.0',
   default_options : [ 'warning_level=3',
                       'c_std=gnu99',
                       'cpp_std=c++17',
diff --git a/spa/meson.build b/spa/meson.build
index 2717b86d6..4e0f96727 100644
--- a/spa/meson.build
+++ b/spa/meson.build
@@ -56,20 +56,9 @@ if not get_option('examples').disabled()
   subdir('examples')
 endif
 
-spa_dep = declare_dependency(
-  include_directories : [spa_inc],
-  version : spaversion,
-  variables : { 'plugindir' : meson.current_build_dir() / 'plugins' }
-)
-
 pkgconfig.generate(filebase : 'lib@0@'.format(spa_name),
   name : 'libspa',
   subdirs : spa_name,
   description : 'Simple Plugin API',
   version : spaversion,
-  extra_cflags : '-D_REENTRANT',
-  variables : ['plugindir=${libdir}/@0@'.format(spa_name)],
-  uninstalled_variables : ['plugindir=${prefix}/spa/plugins'],
-)
-
-meson.override_dependency('lib@0@'.format(spa_name), spa_dep)
+  extra_cflags : '-D_REENTRANT')
diff --git a/src/pipewire/meson.build b/src/pipewire/meson.build
index f66ce8746..5cfaba3f9 100644
--- a/src/pipewire/meson.build
+++ b/src/pipewire/meson.build
@@ -109,12 +109,8 @@ libpipewire = shared_library(pipewire_name, pipewire_sources,
 )
 
 pipewire_dep = declare_dependency(link_with : libpipewire,
-  include_directories : [pipewire_inc, configinc],
-  dependencies : [pthread_lib, atomic_dep, spa_dep],
-  variables : {
-    'moduledir' : meson.current_build_dir() / '..' / 'modules',
-    'confdatadir' : meson.current_build_dir() / '..' / 'daemon',
-  }
+  include_directories : [pipewire_inc, configinc, spa_inc],
+  dependencies : [pthread_lib, atomic_dep, ],
 )
 
 pkgconfig.generate(libpipewire,
@@ -132,6 +128,4 @@ pkgconfig.generate(libpipewire,
   ],
 )
 
-meson.override_dependency('lib@0@'.format(pipewire_name), pipewire_dep)
-
 subdir('extensions')
-- 
2.34.1

